package/libwpe/libwpe.mk

```diff
@@ -10,5 +10,8 @@ LIBWPE_SOURCE = libwpe-$(LIBWPE_VERSION).tar.xz
 LIBWPE_LICENSE = BSD-2-Clause
 LIBWPE_INSTALL_STAGING = YES
 LIBWPE_DEPENDENCIES = libegl libxkbcommon
+LIBWPE_CONF_OPTS += \
+    -DCMAKE_C_FLAGS="$(TARGET_CFLAGS) -D_GNU_SOURCE" \
+    -DCMAKE_CXX_FLAGS="$(TARGET_CXXFLAGS) -D_GNU_SOURCE"
```

Comment:
Compilation was failing without this. The error was related to some CLOCK macro and to fix that error this compiler flag was required

Reply:
I suppose `CLOCK_MONOTONIC` or similar... which is amusing, because `libwpe` does not use the macro (neither do the FDO or RDK backends). I would really prefer to make the CMake files detect whether the flag is needed, and for that we need to know what was the compilation error. Could you try doing a build with this removed and sharing the error? Thanks in advance!

Reply:
../../host/arm-buildroot-linux-gnueabihf/sysroot/usr/include/interface/vcos/pthreads/vcos_platform.h: In function ‘vcos_semaphore_wait_timeout’:
../../host/arm-buildroot-linux-gnueabihf/sysroot/usr/include/interface/vcos/pthreads/vcos_platform.h:297:22: error: ‘CLOCK_REALTIME’ undeclared (first use in this function)
    if (clock_gettime(CLOCK_REALTIME, &ts) == -1)

Reply:
Ouch, this is a bit unfortunate... the package for the Raspberry Pi userland drivers should be the ones defining `_POSIX_C_SOURCE` (or `_GNU_SOURCE`) and putting it into one of its `.pc` files. It seems that there is an issue reported related to this: https://github.com/raspberrypi/userland/issues/316 :upside_down_face: 

I think we need to keep the hack for now, at least until the issue is solved upstream. To use `_CONF_ENV` you can do something like the following:

```make
# Workaround for https://github.com/raspberrypi/userland/issues/316
ifeq ($(BR2_PACKAGE_RPI_USERLAND),y)
LIBWPE_CONF_ENV += CFLAGS='-D_GNU_SOURCE' CXXFLAGS='-D_GNU_SOURCE'
endif
```

Note how we can use a conditional to add the macro definition only when the Raspberry Pi userland drivers are being used.

Reply:
CONF_ENV approach didn't solve the compilation error. I could see both  CFLAGS and CXXFLAGS set before make command but I still got same error.. So I used old workaround.
Let me know if this is ok  

```
# Workaround for https://github.com/raspberrypi/userland/issues/316
ifeq ($(BR2_PACKAGE_RPI_USERLAND),y)
LIBWPE_CONF_OPTS += -DCMAKE_C_FLAGS='$(TARGET_CFLAGS) -D_GNU_SOURCE' \
                    -DCMAKE_CXX_FLAGS='$(TARGET_CFLAGS) -D_GNU_SOURCE'
endif
```

Reply:
I don't want to make you spend more time with this, so let's use the approach you suggest. It's a workaround, so hopefully we can remove it when the userland libraries fix the root cause.